homeassistant/components/bosch_shc/entity.py

```diff
@@ -0,0 +1,105 @@
+"""Bosch Smart Home Controller base entity."""
+from boschshcpy.device import SHCDevice
+
+from homeassistant.helpers.device_registry import async_get as get_dev_reg
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_registry import async_get as get_ent_reg
+
+from .const import DOMAIN
+
+
+async def async_remove_devices(hass, entity, entry_id):
+    """Get item that is removed from session."""
+    ent_registry = get_ent_reg(hass)
+    if entity.entity_id in ent_registry.entities:
+        ent_registry.async_remove(entity.entity_id)
+
+    async def async_get_device_id(hass, device_id):
+        """Get device id from device registry."""
+        dev_registry = get_dev_reg(hass)
+        device = dev_registry.async_get_device(
+            identifiers={(DOMAIN, device_id)}, connections=set()
+        )
+        if device is None:
+            return None
+        return device.id
+
+    dev_registry = get_dev_reg(hass)
+    device_id = async_get_device_id(hass, entity.device_id)
+    if device_id is not None:
+        dev_registry.async_update_device(device_id, remove_config_entry_id=entry_id)
```

Comment:
Why don't we just remove the device entry corresponding to the removed bosch device from the device registry? All connected entity entries and entities will be removed automatically too.